### PR TITLE
feature: CONTAINER_PAUSE state for CRI

### PR DESCRIPTION
**Pull request by zhuangqh**
### Ⅰ. Describe what this PR did

Pause/Unpause container interface are added to CRI, but missing a PAUSE state from `ContainerStatus` or `ListContainer` to figure out the paused container.

### Ⅲ. Why don't you add test cases (unit test/integration test)? (你真的觉得不需要加测试吗？)

cri test here. 
https://github.com/alibaba/cri-tools/pull/11 

### Ⅴ. Special notes for reviews

Signed-off-by: zhuangqh <zhuangqhc@gmail.com>

### CODE REVIEW

File: cri/v1alpha2/cri.go [hunk 1]
```diff
@@ -916,44 +916,7 @@ func (c *CriManager) ContainerStatus(ctx context.Context, r *runtime.ContainerSt
...
-	if container.State.Status == apitypes.StatusRunning {
-		// Container is running.
-		state = runtime.ContainerState_CONTAINER_RUNNING
-	} else {
-		// Container is *not* running. We need to get more details.
-		//		* Case 1: container has run and exited with non-zero finishedAt time
-		//		* Case 2: container has failed to start; it has a zero finishedAt
-		//				  time, but a non-zero exit code.
-		//		* Case 3: container has been created, but not started (yet).
-		finishTime, err := time.Parse(utils.TimeLayout, container.State.FinishedAt)
```
Comment (zhuangqh) [inline cri/v1alpha2/cri.go]: PouchContainer will promise the consistency about `FinishedAt`, `CreatedAt`, `StartedAt` and container state

File: cri/v1alpha2/cri_utils.go [hunk 2]
```diff
@@ -847,21 +847,40 @@ func (c *CriManager) updateCreateConfig(createConfig *apitypes.ContainerCreateCo
 	return nil
 }
 
-func toCriContainerState(status apitypes.Status) runtime.ContainerState {
-	switch status {
-	case apitypes.StatusRunning:
-		return runtime.ContainerState_CONTAINER_RUNNING
-	case apitypes.StatusExited:
-		return runtime.ContainerState_CONTAINER_EXITED
+func toCriContainerState(state *apitypes.ContainerState) (criState runtime.ContainerState, reason string) {
```
Comment (HusterWan) [inline cri/v1alpha2/cri_utils.go]: Why just define the return type `runtime.ContainerState, string` and return the value in the function?

Comment (zhuangqh) [reply cri/v1alpha2/cri_utils.go]: Did you mean the named return value?

Comment (HusterWan) [reply cri/v1alpha2/cri_utils.go]: yes

**Comment by HusterWan** [comment]
LGTM

**Pull request merged**